code/game/objects/items/RSF.dm

```diff
@@ -8,178 +8,200 @@ RSF
-			to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
-			icon_state = "rsf_empty"
-			return
-	else if (matter < 1)
-		to_chat(user, "<span class='warning'>\The [src] doesn't have enough matter left.</span>")
-		icon_state = "rsf_empty"
-		return
-
-	var/turf/T = get_turf(A)
-	playsound(src.loc, 'sound/machines/click.ogg', 10, TRUE)
-	switch(mode)
-		if(1)
-			to_chat(user, "<span class='notice'>Dispensing Drinking Glass...</span>")
-			new /obj/item/reagent_containers/food/drinks/drinkingglass(T)
-			use_matter(20, user)
-		if(2)
-			to_chat(user, "<span class='notice'>Dispensing Paper Sheet...</span>")
-			new /obj/item/paper(T)
-			use_matter(10, user)
-		if(3)
-			to_chat(user, "<span class='notice'>Dispensing Pen...</span>")
-			new /obj/item/pen(T)
-			use_matter(50, user)
-		if(4)
-			to_chat(user, "<span class='notice'>Dispensing Dice Pack...</span>")
-			new /obj/item/storage/pill_bottle/dice(T)
-			use_matter(200, user)
-		if(5)
-			to_chat(user, "<span class='notice'>Dispensing Cigarette...</span>")
-			new /obj/item/clothing/mask/cigarette(T)
-			use_matter(10, user)
-
+///A helper proc. checks to see if we can afford the amount of charge that is passed, and if we can docs the charge from our base, and returns TRUE. If we can't we return FALSE
 /obj/item/rsf/proc/use_matter(charge, mob/user)
-	if (iscyborg(user))
+	if(iscyborg(user))
 		var/mob/living/silicon/robot/R = user
-		R.cell.charge -= charge
+		var/end_charge = R.cell.charge - charge
+		if(end_charge >= 0)
+			R.cell.charge = end_charge
+			return TRUE
+		to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
 	else
-		matter--
-		to_chat(user, "<span class='notice'>The RSF now holds [matter]/30 fabrication-units.</span>")
-
-/obj/item/cookiesynth
+		if(matter - 1 >= 0)
+			matter--
+			to_chat(user, "<span class='notice'>\The [src] now holds [matter]/[max_matter] [discriptor].</span>")
+			return TRUE
+		to_chat(user, "<span class='warning'>\The [src] doesn't have enough [discriptor] left.</span>")
+	icon_state = spent_icon_state
+	return FALSE
+
+///Helper proc that iterates through all the things we are allowed to spawn on, and sees if the passed atom is one of them
+/obj/item/rsf/proc/is_allowed(atom/to_check)
+	for(var/sort in allowed_surfaces)
+		if(istype(to_check, sort))
+			return TRUE
+	return FALSE
+
+/obj/item/rsf/cookiesynth
 	name = "Cookie Synthesizer"
 	desc = "A self-recharging device used to rapidly deploy cookies."
-	icon = 'icons/obj/tools.dmi'
 	icon_state = "rcd"
-	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
-	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
-	var/matter = 10
-	var/toxin = 0
+	spent_icon_state = "rcd"
+	var/toxin = FALSE
 	var/cooldown = 0
 	var/cooldowndelay = 10
-	w_class = WEIGHT_CLASS_NORMAL
+	max_matter = 10
+	cost_by_item = list()
+	dispense_cost = 100
+	discriptor = "cookie-units"
+	action_type = "Fabricates"
```

Comment:
Document new variables, and put the redefined variables before the new ones.

Reply:
Can do